static/app/components/events/eventStatisticalDetector/eventDifferentialFlamegraph.tsx

```diff
@@ -316,6 +369,267 @@ function DifferentialFlamegraphTransactionToolbar(
   );
 }
 
+interface PaginationReducerState {
+  page: number;
+  pageCount: number;
+  pageSize: number;
+}
+
+type PaginationReducerAction =
+  | {type: 'next'}
+  | {type: 'previous'}
+  | {list: any[]; pageSize: number; type: 'initialize'};
+
+function paginationReducer(
+  state: PaginationReducerState,
+  action: PaginationReducerAction
+): PaginationReducerState {
+  switch (action.type) {
+    case 'initialize': {
+      return {
+        page: 0,
+        pageCount: Math.ceil(action.list.length / action.pageSize),
+        pageSize: action.pageSize,
+      };
+    }
+    case 'next':
+      return {
+        ...state,
+        page: Math.min(state.page + 1, state.pageCount),
+      };
+    case 'previous':
+      return {
+        ...state,
+        page: Math.max(state.page - 1, 0),
+      };
+    default:
+      return state;
+  }
+}
+
+interface DifferentialFlamegraphChangedFunctionsProps {
+  flamegraph: DifferentialFlamegraphModel;
+  functions: DifferentialFlamegraphModel['increasedFrames'];
+  loading: boolean;
+  makeFunctionLink: (frame: FlamegraphFrame) => LocationDescriptor;
+  subtitle: string;
+  title: string;
+}
+function DifferentialFlamegraphChangedFunctions(
+  props: DifferentialFlamegraphChangedFunctionsProps
+) {
+  const [state, dispatch] = useReducer(paginationReducer, {
+    page: 0,
+    pageSize: 0,
+    pageCount: 0,
+  });
+
+  useEffect(() => {
+    dispatch({
+      list: props.functions,
+      pageSize: 5,
+      type: 'initialize',
+    });
+  }, [props.functions]);
+
+  const onPreviousPaginationClick = useMemo(() => {
+    if (state.page === 0) {
+      return undefined;
+    }
+    return () => dispatch({type: 'previous'});
+  }, [state.page]);
+
+  const onNextPaginationClick = useMemo(() => {
+    if (state.page + 1 === state.pageCount) {
+      return undefined;
+    }
+    return () => dispatch({type: 'next'});
+  }, [state.page, state.pageCount]);
```

Comment:
The guards here are just to disable the previous/next buttons right? As the reducer themselves already handle the events for the first/last page.

Reply:
Yup, correct!